2_process/src/sites_along_waterbody.R

```diff
@@ -19,10 +18,39 @@ sites_along_waterbody <- function(sw_sites_sf, waterbody_sf, lake_waterbody = FA
       summarize(geometry = sf::st_union(geometry)) %>%
       sf::st_buffer(dist = units::set_units(250, m))
   }
-  filtered_sites <- st_join(sw_sites, waterbody_buffered, left = FALSE) %>%
+  filtered_sites <- st_join(sites_sf, waterbody_buffered, left = FALSE) %>%
     pull(site_no) %>%
     unique()
   
   return(filtered_sites)
   
 }
+
+#' @title add_stream_order
+#' @description function that joins nwis site info to nwis sw data and adds stream order col
+#' @param nwis_sw_data nwis sw data pulled from dataRetrieval::readNWISdata() or similar dataRetrieval functions 
+#' @param sites_sf sf object of sf sites and measurements
+#' @param join_site_col col to site col to join sites_sf with nwis_sw_data. defaults is site_no as that is expected to be common between both dfs
+#' @param sites_along_streamorder3 vector of site_no that are along stream order 3 streams. Output of sites_along_waterbody()    
+#' @param sites_along_lake vector of sites that are adjacent to saline lakes. Output of sites_along_waterbody
+
+add_stream_order <- function(nwis_sw_data, sites_sf, join_site_col = 'site_no',  sites_along_streamorder3, sites_along_lake){
+  
+  nwis_sw_data %>%
+    left_join(sites_sf, by = join_site_col) %>%
+    ## filtering sites that are surface level stream (all ST), lake (LK), or wetland (WE) sites
+    filter(site_tp_cd %in% c('LK','WE') | grepl('ST',site_tp_cd)) %>% 
```

Comment:
Wondering if we really want to filter out data in the process of adding a column.This is omitting a spring site `SP` and some NA sites. The NAs seem to be empty rows,.